flexmeasures/api/v2_0/implementations/sensors.py

```diff
@@ -46,8 +46,16 @@
 @type_accepted("PostPriceDataRequest")
 @units_accepted("price", "EUR/MWh", "KRW/kWh")
 @assets_required("market")
-@optional_horizon_accepted()
-@optional_prior_accepted()
+@optional_horizon_accepted(
+    infer_missing=True
+    if current_app.config.get("FLEXMEASURES_MODE", "") == "play"
```

Comment:
@Flix6x The use of `current_app` "outside" the function like this is not possible, as there is no application context at that point. I learned that from all our tests falling, as importing this file breaks.

We have to find a different way. If, for play mode, `infer_missing` is always True for horizons ad False for prior, maybe it could go into the respective validators.

Reply:
Thanks for finding out. I'll move the logic back into the validators, and devise of some other way of letting v1.x and v2 endpoints choose to use this logic. Alas, `infer_missing` is not always True for horizons in play mode. Specifically, it is False for the postPrognosis endpoint v1.x, even in play mode. I could make the boolean optional, and set the default to None implying inference, so endpoints can be explicit or let the validator choose a default depending on server mode.

For the horizon validator:

- v1.x postPrognosis: `infer_missing=False`
- v1.x postOther: `infer_missing=True`
- v2.x postPrognosis: `infer_missing=False`
- v2.x postOther: `infer_missing=None` -> True if in play, False otherwise

For the prior validator:

- v1.x doesn't use this validator
- v2.x postPrognosis and postOther: `infer_missing=None` -> False if in play, True otherwise

So to be clear, the differences between v1.x and v2 are that:
- The only time a user needs to be bothered with communicating the timing of their beliefs is for the postPrognosis endpoint: for v1.x this is **in all modes**, whereas for v2, this is **in play mode only**.
- The inference policy for v1.x is **beliefs formed right after the fact**, whereas for v2, this is **beliefs formed when communicated to FM**.